examples/GainPlugin/PluginProcessor.cpp

```diff
@@ -28,158 +28,170 @@ AudioProcessorValueTreeState::ParameterLayout createParameterLayout()
+    return false;
+   #endif
+}
+
+double GainPluginAudioProcessor::getTailLengthSeconds() const
+{
+    return 0.0;
+}
+
+int GainPluginAudioProcessor::getNumPrograms()
+{
+    return 1;   // NB: some hosts don't cope very well if you tell them there are 0 programs,
+                // so this should be at least 1, even if you're not really implementing programs.
+}
+
+int GainPluginAudioProcessor::getCurrentProgram()
+{
+    return 0;
+}
+
+void GainPluginAudioProcessor::setCurrentProgram (int index)
+{
+}
+
+const String GainPluginAudioProcessor::getProgramName (int index)
+{
+    return {};
+}
+
+void GainPluginAudioProcessor::changeProgramName (int index, const String& newName)
+{
+}
+
+//==============================================================================
+void GainPluginAudioProcessor::prepareToPlay (double sampleRate, int samplesPerBlock)
+{
+    gain.reset(sampleRate, 0.02);
+}
+
+void GainPluginAudioProcessor::releaseResources()
+{
+    // When playback stops, you can use this as an opportunity to free up any
+    // spare memory, etc.
+}
+
+#ifndef JucePlugin_PreferredChannelConfigurations
+bool GainPluginAudioProcessor::isBusesLayoutSupported (const BusesLayout& layouts) const
+{
+  #if JucePlugin_IsMidiEffect
+    ignoreUnused (layouts);
+    return true;
+  #else
+    // This is the place where you check if the layout is supported.
+    // In this template code we only support mono or stereo.
+    if (layouts.getMainOutputChannelSet() != AudioChannelSet::mono()
+     && layouts.getMainOutputChannelSet() != AudioChannelSet::stereo())
+        return false;
+
+    // This checks if the input layout matches the output layout
+   #if ! JucePlugin_IsSynth
+    if (layouts.getMainOutputChannelSet() != layouts.getMainInputChannelSet())
+        return false;
+   #endif
+
+    return true;
+  #endif
+}
+#endif
+
+void GainPluginAudioProcessor::processBlock (AudioBuffer<float>& buffer, MidiBuffer& midiMessages)
+{
+    ScopedNoDenormals noDenormals;
+    auto totalNumInputChannels  = getTotalNumInputChannels();
+    auto totalNumOutputChannels = getTotalNumOutputChannels();
+
+    // In case we have more outputs than inputs, this code clears any output
+    // channels that didn't contain input data, (because these aren't
+    // guaranteed to be empty - they may contain garbage).
+    // This is here to avoid people getting screaming feedback
+    // when they first compile a plugin, but obviously you don't need to keep
+    // this code if your algorithm always overwrites all the output channels.
+    for (auto i = totalNumInputChannels; i < totalNumOutputChannels; ++i)
+        buffer.clear (i, 0, buffer.getNumSamples());
 
     // Our intense dsp processing
-    gain.setValue(*params.getRawParameterValue("MainGain"));
+    gain.setValue(*params.getRawParameterValue("MainGain"));
     gain.applyGain(buffer, buffer.getNumSamples());
 
+    if (auto *muteParam  = dynamic_cast<AudioParameterBool*>(params.getParameter("MainMute")))
```

Comment:
This is the only significant change. Adding the MainMute param and the wild dsp.